### DISCUSSION TITLE
I failed in stalling the WSL, please help me

### DISCUSSION THREAD

**Discussion by lynxwow** [microsoft/WSL#9860] [General]
### Windows Version

Microsoft Windows [Version 10.0.22000.675]

### WSL Version

0.0.0.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Repro Steps

wsl --install

### Expected Behavior

Installing...This may take a few minutes.

### Actual Behavior

Copyright (c) Microsoft Corporation. All rights reserved.

Usage: wsl.exe [parameter] [options...] [command line]

Parameter for running Linux binaries:

If no command line is provided, wsl.exe launches the default shell.

--exec, -e <CommandLine>
    Execute the specified command without using the default Linux shell.

--
    Pass the remaining command line as is.
Options:
--cd <Directory>
Set the specified directory as the current working directory.
If ~ is used, the home path of the Linux user will be used. If the path
starts with a / character, it will be interpreted as an absolute Linux path.
Otherwise, the value must be an absolute Windows path.

--distribution, -d <DistributionName>
    Run the specified distribution.

--user, -u <UserName>
    Run as the specified user.

--system
    Launches the shell for the system distribution.
Parameter for managing Linux distributions for Windows Subsystem for Linux:

--help
    Display usage information.

--install [Options]
    Install additional Linux distributions for Windows Subsystem for Linux.
    To obtain a list of valid distributions, use "wsl --list --online".

    Options:
        --distribution, -d [argument]
            Download and install the distribution by name.

            Argument:
                Valid distribution name (case insensitive).

            Example:
                wsl --install -d Ubuntu
                wsl --install --distribution Debian

--set-default-version <Version>
    Change the default installation version for new distributions.

--shutdown
    Immediately terminates all running distributions and the WSL 2
    lightweight utility virtual machine.

--status
    Display the status of the Windows Subsystem for Linux.

--update [Options]
    Updates the WSL 2 kernel to the latest version if no options are specified.

    Options:
        --rollback
            Roll back to a previous version of the WSL 2 kernel.
Parameter for managing distributions within Windows Subsystem for Linux:

--export <DistributionName> <FileName>
    Export the distribution to a tar file.
    For standard output, the file name can be -.

--import <DistributionName> <InstallLocation> <FileName> [Options]
    Import the specified tar file as a new distribution.
    For standard input, the file name can be -.

    Options:
        --version <Version>
            Specify the version to use for the new distribution.

--list, -l [Options]
    List distributions.

    Options:
        --all
            List all distributions, including those currently being installed or uninstalled.

        --running
            List only running distributions.

        --quiet, -q
            Show only distribution names.

        --verbose, -v
            Show detailed information for all distributions.

        --online, -o
            Show a list of available distributions for installation using "wsl --install".

--set-default, -s <DistributionName>
    Set the specified distribution as the default.

--set-version <DistributionName> <Version>
    Change the version of the specified distribution.

--terminate, -t <DistributionName>
    Terminate the specified distribution.

--unregister <DistributionName>
    Unregister the distribution and delete the root filesystem.

--mount <Disk>
    Connect and mount a physical disk in all WSL2 distributions.

    Options:
        --bare
            Connect the disk to WSL2 without mounting it.

        --type <Type>
            The filesystem to be used when mounting the disk. Defaults to ext4 if not specified.

        --options <Options>
            Other mount options.

        --partition <Index>
            The index of the partition to be mounted. Defaults to the entire disk if not specified.

--unmount [Disk]
    Unmount and detach the disk from all WSL2 distributions.
    If called without parameters, all disks will be unmounted and detached.

### Diagnostic Logs

i remembered intalling it once, but the Ubuntu can't be found in my computer and when i tried to reinstall it and ran the same code, it only sprung out these words.

I tried to search for its menu and github and on the internet but in vain.

and i tried 
https://blog.csdn.net/qq_43780850/article/details/124635249

https://blog.csdn.net/qq_43780850/article/details/124633611?csdn_share_tail=%7B%22type%22:%22blog%22,%22rType%22:%22article%22,%22rId%22:%22124633611%22,%22source%22:%22qq_43780850%22%7D&ctrtid=k5Lqc

but either did it not work or i failed to found or install the Windows Security on my Win11 

I really want to know how can i succeed in it, please help me
Thanks for your guys!

**Comment by tobiaspanknin** [comment]
The output suggests WSL2 is installed. It gives you some help on the available commands.

To verify if your old Ubuntu distribution or any other distribution is still installed issue the command
`wsl --list` 

If any distribution is listed there you can run it by issueing the command 
`wsl -d '<copy & paste the exact distribution name here between the hypens>' `

otherwise you can get a list of distributions available for installation using the command 
`wsl --list --online`
then install the distribution of your cjoice using the command
`wsl --install '<copy & paste the distribution you want to install as listed by the list command>'`

**Comment by lynxwow** [comment]
Thank you for your kind reply! but actually when i typed wsl --list：it turns out to be :

------------
The Windows Subsystem for Linux (WSL) on Linux does not have a distribution installed by default. You can install a distribution by accessing the Microsoft Store.
------------

and MS is not available for my computer , so i tried 

-----------
wsl --install -'Ubuntu'
-----------

but  it seems falling asleep at the switch, just couldn't react in any way

**Comment by tobiaspanknin** [comment]
Try without the - before the Distro name. Also if there are no spaces in the distro name you don't need the hyphens so it would just be `wsl --install Ubuntu` in your case

**Comment by lynxwow** [comment]
thanks again!  but it repeated the words instead of working. Maybe my computer is sleeping or even dreaming right now,oh no. I am going crazy

PS C:\Users\ASUS> wsl --install Ubuntu
Copyright (c) Microsoft Corporation. All rights reserved.

Usage: wsl.exe [parameter] [option...] [command line]

Parameters for running Linux binaries:

If no command line is provided, wsl.exe launches the default shell.

--exec, -e <command line>
    Execute the specified command without using the default Linux shell.

--
    Passes the remaining command line as is.
......and so on

**Comment by tobiaspanknin** [comment]
what is the output of the `wsl --list` command ?

**Comment by lynxwow** [comment]
the output were as follows:The Windows Subsystem for Linux (WSL) on Linux does not have a distribution installed by default. You can install a distribution by accessing the Microsoft Store.
        
    
        
            
        
    
         
        
            
                        ***@***.***
            
        
     
    
  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            Tobias ***@***.***>
            
        
      
      
        
         Date 
        
        
        3/28/2023 22:31
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [microsoft/WSL] I failed in stalling the WSL, please help me (Issue #9858)
        
      
    
  

what is the output of the wsl --list command ?
